src/reflect.ts

```diff
@@ -13,7 +13,11 @@ export type Reflect =
   | { tag: 'symbol' } & Runtype<symbol>
   | { tag: 'literal'; value: LiteralBase } & Runtype<LiteralBase>
   | { tag: 'array'; element: Reflect } & Runtype<unknown[]>
+  | { tag: 'readonlyArray'; element: Reflect } & Runtype<ReadonlyArray<unknown>>
   | { tag: 'record'; fields: { [_: string]: Reflect } } & Runtype<{ [_ in string]: unknown }>
+  | { tag: 'readonlyRecord'; fields: { [_: string]: Reflect } } & Runtype<
+      { readonly [_ in string]: unknown }
+    >
```

Comment:
The point of what I was saying was not to introduce new core types like this.

Reply:
Oh. I didn't fix any of the other files to the new style yet, since I didn't want to invest that effort again before hearing of your opinion on the other implementation.

Please check the record.ts and array.ts files, those are the only ones I updated in my second commit